userspace/libsinsp/sinsp.h

```diff
@@ -1209,6 +1205,18 @@ VISIBILITY_PRIVATE
 	friend class sinsp_network_interfaces;
 
 	template<class TKey,class THash,class TCompare> friend class sinsp_connection_manager;
+
+#ifdef SYSDIG_TEST
+protected:
```

Comment:
Seems odd for it to be both `SYSDIG_TEST` and `protected`?

Reply:
even in test, I only want the derived class to inject that data